src/main/java/com/github/sgov/server/service/WorkspaceService.java

```diff
@@ -109,18 +107,10 @@ public URI ensureVocabularyExistsInWorkspace(
             return vocabularyContextUri;
         }
 
-        if (vocabularyService.getVocabulariesAsContextDtos().stream()
-            .noneMatch(vc ->
-                vc.getBasedOnVersion().equals(vocabularyUri)
-            )
-        ) {
-            if (vocabularyContextDto.getLabel() == null) {
-                throw NotFoundException.create("Vocabulary", vocabularyUri);
-            }
-            return createVocabularyContext(workspace, vocabularyContextDto);
-        } else {
-            return loadVocabularyContextFromCache(workspace, vocabularyUri);
+        if (vocabularyContextDto.getLabel() == null) {
```

Comment:
I don't understand how the vocabulary is loaded from cache now, if it has been already published before.

Reply:
@psiotwo please sync comment with implementation